src/dotnet/UI.Blazor/Services/PersistentState.cs

```diff
@@ -0,0 +1,70 @@
+using System.Reactive.Linq;
+
+namespace ActualChat.UI.Blazor.Services;
+
+public interface IPersistentState<T> : IMutableState<T>, IAsyncDisposable
+{ }
+public class PersistentState<T>: MutableState<T>, IPersistentState<T>
+{
+    private readonly object _lock = new ();
+    private readonly CancellationTokenSource _cts;
+    private IDisposable? _updatedObservableCompletion;
+    private Task _lastPersistTask;
+
+    protected Func<T, ISessionCommand> PersistCommandFactory { get; }
```

Comment:
I would say this is kinda complicated. I rather use `string Key` here & persist any of such states via `IPersistentStateStore.Persist`, which is typically a scoped service relying on whatever makes sense to do that.

Reply:
I don't like the idea that everything as a key-value pair